core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -481,8 +482,25 @@ public void dayOfYear() {
     assertEquals(integerValue(220), eval(expression));
   }
 
+  @Test
+  public void testDayOfYearWithTimeType() {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+    FunctionExpression expression = DSL.day_of_year(
+        functionProperties,
+        DSL.literal(new ExprTimeValue("12:23:34")));
+
+    assertEquals(INTEGER, eval(expression).type());
+    assertEquals(LocalDate.now(
+            functionProperties.getQueryStartClock()).getDayOfYear(),
+        eval(expression).integerValue());
+    assertEquals("day_of_year(TIME '12:23:34')", expression.toString());
+  }
+
   public void testDayOfYearWithUnderscores(String date, int dayOfYear) {
```

Comment:
`@Test`

Reply:
Function renamed instead in 85c93e677ba64de401fa4b4115acfe428005e003 as it is a helper function